### ISSUE TITLE
"DPANIC	odd number of arguments passed as key-value pairs" for logging for multiple triggers

### ISSUE THREAD

**Issue by thomasmRavn** [kedacore/charts#430]
Some of our scaled objects have multiple triggers. When upgrading to keda 2.10.1 the operator throws the error message below, when trying to resolve the prometheus trigger.
```
2023-04-17T10:28:31Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "ScaledObject": {"name":"ku-tenant-smoke-ku-api-scaledobject","namespace":"ku-tenant-smoke"}, "namespace": "ku-tenant-smoke", "name": "ku-tenant-smoke-ku-api-scaledobject", "reconcileID": "98a06eb3-5106-49e3-8d23-777e58ec8f26"}
2023-04-17T10:28:31Z	DPANIC	odd number of arguments passed as key-value pairs for logging	{"ignored key": "prometheus"}
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).checkTriggers
	/workspace/controllers/keda/scaledobject_controller.go:358
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).reconcileScaledObject
	/workspace/controllers/keda/scaledobject_controller.go:223
github.com/kedacore/keda/v2/controllers/keda.(*ScaledObjectReconciler).Reconcile
	/workspace/controllers/keda/scaledobject_controller.go:175
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:122
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:323
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/workspace/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:235
2023-04-17T10:28:31Z	INFO	metricName is deprecated and will be removed in v2.12, please do not set it anymore (used in %q)
```
The metricName we are aware of the deprecation, and have a plan to remove but I don't believe this is causing the error. Because metricName is set for all( 23) scaledobjects but the error only appears above the line reconciling for <service with 2 triggers defined>

## Expected Behavior
no DPANIC message sent. I think this is telling me that keda is not logging correctly so not sure what the correct log line should look like but something like
```
2023-04-17T10:28:22Z	INFO	Reconciling ScaledObject	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "ScaledObject": {"name":"ku-policy-command-service-scaledobject","namespace":"ku-tenant-smoke"}, "namespace": "ku-tenant-smoke", "name": "ku-policy-command-service-scaledobject", "reconcileID": "5dc9491b-045a-47e6-a8dc-b0cc0594a8d6"}
```

## Actual Behavior
Lots of DPANIC messages in the logs.

## Steps to Reproduce the Problem

  1.Deploy a scaled object with more than 1 trigger defined
  ```
  ---
  apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: ku-support-api-scaledobject
  namespace: ku-tenant-dev
spec:
  scaleTargetRef:
    name: ku-support-api                    # Mandatory. Must be in the same namespace as the ScaledObject
  pollingInterval: 15
  cooldownPeriod:  300                              # Optional. Default: 300 seconds
  minReplicaCount: 1
  maxReplicaCount: 10
  advanced:                                          # Optional. Section to specify advanced options
    horizontalPodAutoscalerConfig:                   # Optional. Section to specify HPA related options
      behavior:                                      # Optional. Use to modify HPA's scaling behavior
        scaleDown:
          stabilizationWindowSeconds: 120
  triggers:
    - type: prometheus
      metadata:
        # Required
        serverAddress: http://prom-prometheus.ravn-metrics.svc.cluster.local:9090
        metricName: istio_request_rate
        threshold: "75"
        query: (sum(rate(http_server_requests_seconds_sum{container="ku-support-api",namespace="ku-tenant-dev"}[5m]))/sum(rate(http_server_requests_seconds_count{container="ku-support-api",namespace="ku-tenant-dev"}[5m])))*1000
    - type: cpu
      metadata:
        # Required
        type: Utilization
        value: "80"
```
  2. tail the logs of the keda operator
  `kubectl -n <keda ns> logs -f --since 1m deploy/keda-operator`
  3. Should see the error message DPANIC appear fairly aggressively, since this will happen every time keda tries to reconcile the scaled object

## Specifications

  - **KEDA Version:** kedacore/keda:2.10.0
  - **Platform & Version:**  not sure what you mean here, the chart is 2.10.1
  - **Kubernetes Version:** v1.25.5
  - **Scaler(s):** prometheus

**Comment by thomasmRavn** [comment]
I should have said in the ticket sorry but this isn't an issue with keda chart 2.9.4
same scaledobject.

**Comment by zroubalik** [comment]
Hi, this has been fixed in KEDA 2.10.1 (chart 2.10.2)

**Comment by thomasmRavn** [comment]
oh awesome! Sorry for wasting your time
